lucene/suggest/src/java/org/apache/lucene/search/suggest/InputIterator.java

```diff
@@ -34,7 +34,7 @@
 public interface InputIterator extends BytesRefIterator {
 
   /** A term's weight, higher numbers mean better suggestions. */
```

Comment:
Maybe add javadocs explaining what `null` means?  Though, why do we need to allow for `null`?  Shouldn't the iterator not return a suggestion that has no weight?

Reply:
Hi Michael,
The reason to allow for null at the InputIterator level is to distinguish it from an explicit 0 weight.
In the DocumentDictionary this translates in differentiating when  the weight field was missing for the original document ( NULL ) in opposition to when the weight field was present and with 0 value.
At this level we just want to ensure that the same behavior is maintained when we build the auxiliary index : 
i.e. if the weight field was missing for the original document, I want it to be null for the auxiliary index as well.
How the different suggesters implementation will use this to return a suggestion score, I think will depend on a case by case scenario.
Did I misunderstand anything here ?